File: backend/semgrep_rules/tests/python/kisa_sw_36_error_message_exposure.py
from django.http import HttpResponse
def unsafe():
    try:
        1 / 0
    except Exception as e:
        # ruleid: kisa.sw36.python.exception-detail-to-response
        return HttpResponse(str(e))
def safe():
    try:
        1 / 0
    except Exception:
        # ok: kisa.sw36.python.exception-detail-to-response
        return HttpResponse("internal error", status=500)

File: backend/semgrep_rules/tests/javascript/kisa_sw_36_error_message_exposure.js
function unsafe(err, res) {
  // ruleid: kisa.sw36.javascript.exception-detail-to-response
  res.status(500).send(err.stack);
}
function safe(err, res) {
  console.error(err);
  // ok: kisa.sw36.javascript.exception-detail-to-response
  res.status(500).send("internal error");
}
